File: food-order-client/src/types/order.ts
import type { OrderStatus, PaymentMethod } from '../utils/constants';

export interface OrderItem {
  id: number;
  productId: number;
  productNameEn: string;
  productNameAr: string;
  quantity: number;
  unitPrice: number;
  imageUrl: string;
}

export interface Order {
  id: number;
  userId: number;
  customerName: string;
  totalAmount: number;
  status: OrderStatus;
  paymentMethod: PaymentMethod;
  deliveryAddress: string;
  phoneNumber: string;
  notes?: string;
  createdAt: string;
  updatedAt: string;
  items: OrderItem[];
}

export interface CreateOrderRequest {
  deliveryAddress: string;
  phoneNumber: string;
  notes?: string;
  paymentMethod: PaymentMethod;
}

File: FoodOrderAPI/Models/Order.cs
using System;
using System.Collections.Generic;

namespace FoodOrderAPI.Models
{
    public enum OrderStatus
    {
        Pending,
        Preparing,
        OutForDelivery,
        Delivered
    }

    public enum PaymentMethod
    {
        COD,
        Online
    }

    public class Order
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User? User { get; set; }
        public decimal TotalAmount { get; set; }
        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.COD;
        public string DeliveryAddress { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string? Notes { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
